### PR TITLE
Feature/otf 1500 column comparison

### PR DESCRIPTION
**Pull request by jenbaldwin**

### CODE REVIEW

File: api/src/main/java/org/apache/iceberg/expressions/StrictMetricsEvaluator.java [hunk 1]
```diff
@@ -218,6 +218,11 @@ public <T> Boolean lt(BoundReference<T> ref, Literal<T> lit) {
       return ROWS_MIGHT_NOT_MATCH;
     }
 
+    //    @Override
```
Comment (jenbaldwin) [inline api/src/main/java/org/apache/iceberg/expressions/StrictMetricsEvaluator.java:221]: not implemented yet

File: api/src/main/java/org/apache/iceberg/expressions/BoundTermPredicate.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    return true;
+  }
+
+  @Override
+  public BoundTermPredicate<T> asTermPredicate() {
+    return this;
+  }
+
+  @Override
+  public boolean test(T value) { // TODO
```
Comment (jenbaldwin) [inline api/src/main/java/org/apache/iceberg/expressions/BoundTermPredicate.java:45]: need to test this

File: core/src/main/java/org/apache/iceberg/BaseEntriesTable.java [hunk 3]
```diff
@@ -217,6 +237,11 @@ public <T> Boolean eq(BoundReference<T> ref, Literal<T> lit) {
         return ROWS_MIGHT_MATCH;
       }
 
+      @Override
+      public <T> Boolean eq(BoundReference<T> ref, BoundReference<T> ref2) { // TODO
```
Comment (jenbaldwin) [inline core/src/main/java/org/apache/iceberg/BaseEntriesTable.java:241]: not implemented this yet, not sure how to test

**Comment by nastra** [comment]
@jenbaldwin could you please add some context to the problem that this PR is trying to solve? Also what does OTF-1500 stand for?

**Comment by jenbaldwin** [comment]
didn't intend to open draft PR to main, wip

**Pull request closed without merging**